examples/tilted_bottom_boundary_layer.jl

```diff
@@ -95,10 +95,19 @@ coriolis = ConstantCartesianCoriolis(f = 1e-4, rotation_axis = ĝ)
 # _perturbations_ away from the constant density stratification by imposing
 # a constant stratification as a `BackgroundField`,
 
-B_field = BackgroundField(constant_stratification, parameters=(; ĝ, N² = 1e-5))
+N² = 1e-5 # s⁻¹ # background stratification
+B_field = BackgroundField(constant_stratification, parameters=(; ĝ, N² = N²))
 
 # where ``N^2 = 10^{-5} \rm{s}^{-2}`` is the background buoyancy gradient.
 
+# Because the bottom boundary condition is that there must be zero *total* diffusive
+# flux across the seafloor, the default `FluxBoundaryCondition()` is insufficient;
+# instead, the boundary condition on the perturbation flux must take the background
+# into account,
```

Comment:
For this I think we should write down the boundary condition that we want to prescribe (I think it's important to make clear that the boundary condition on the total buoyancy is a _choice_ --- once that is decided, then one has to do the math to _compute_ what the boundary condition should be on the perturbation field). This could be important for users that want to use a different boundary condition than zero diffusive flux across the seafloor, and also just pedagogical...

Reply:
How's this: https://github.com/CliMA/Oceananigans.jl/pull/3581/commits/d3dec7f4c0849940d183eebc58a70b4cf5bb997b#diff-a8b1666e910f9c11760ffa5cd0c5b1b0964c32a97b4908aeeb096067a990fac9R101-R111